compiler-lib/src/asciifile/span.rs

```diff
@@ -274,6 +274,36 @@ impl<'span, 'file> Iterator for LineIterator<'span, 'file> {
     }
 }
 
+/// Iterator adapter that adds line numbers to a `LineIterator`
+///
+/// ```
+/// use compiler_lib::asciifile::{AsciiFile, Position, Span};
```

Comment:
does `cargo test` test that rustdoc examples compile?

Reply:
yes.